src/main/java/com/bugsnag/Configuration.java

```diff
@@ -97,6 +102,54 @@ public boolean shouldAutoCaptureSessions() {
         return autoCaptureSessions.get();
     }
 
+    /**
+     * Set the endpoints to send data to. By default we'll send error reports to
+     * https://notify.bugsnag.com, and sessions to https://sessions.bugsnag.com, but you can
+     * override this if you are using Bugsnag Enterprise to point to your own Bugsnag endpoint.
+     *
+     * Please note that it is recommended that you set both endpoints. If the notify endpoint is
+     * missing, an exception will be thrown. If the session endpoint is missing, a warning will be
+     * logged and sessions will not be sent automatically.
+     *
+     * Note that if you are setting a custom {@link Delivery}, this method should be called after
+     * the custom implementation has been set.
+     *
+     * @param notify the notify endpoint
+     * @param sessions the sessions endpoint
+     *
+     * @throws IllegalArgumentException if the notify endpoint is empty or null
+     */
+    public void setEndpoints(String notify, String sessions) throws IllegalArgumentException {
+        if (notify == null || notify.length() == 0) {
+            throw new IllegalArgumentException("Notify endpoint cannot be empty or null.");
+        } else {
+            if (delivery instanceof HttpDelivery) {
+                ((HttpDelivery) delivery).setEndpoint(notify);
+            } else {
+                LOGGER.warn("Delivery is not instance of HttpDelivery, cannot set notify endpoint");
+            }
+        }
+
+        boolean invalidSessionsEndpoint = sessions == null || sessions.length() == 0;
+        String sessionEndpoint = null;
+
+        if (invalidSessionsEndpoint) {
+            LOGGER.warn("The session tracking endpoint has not been"
+                    + " set. Session tracking is disabled");
+            this.autoCaptureSessions.set(false);
+        } else {
+            sessionEndpoint = sessions;
+        }
+
+        if (sessionDelivery instanceof HttpDelivery) {
+            // sessionEndpoint may be invalid (i.e. null) here, which should result in the default
+            // HttpDelivery throwing a MalformedUrlException which prevents delivery.
```

Comment:
are we sure this is what we want to be doing in this case? It seems less optimal to use invalid data and rely on exceptions being thrown to prevent delivery